astropy/cosmology/tests/test_funcs.py

```diff
@@ -214,7 +214,7 @@ def test_z_at_value_unconverged(method):
 @pytest.mark.parametrize('cosmo', [Planck13, Planck15, Planck18, WMAP5, WMAP7, WMAP9,
                                    core.LambdaCDM, core.FlatLambdaCDM, core.wpwaCDM, core.w0wzCDM,
                                    core.wCDM, core.FlatwCDM, core.w0waCDM, core.Flatw0waCDM])
-def test_z_at_value_roundtrip(cosmo):
+def test_z_at_value_roundtrip(cosmo, recwarn):
```

Comment:
Is this a proper fix or a temporary workaround?

Reply:
If I can reproduce the error I can tell for sure, but I suspect that it's a 0 division warning arising from the ``self._comoving_transverse_distance_z1z2(0, z)`` in ``comoving_transverse_distance(self, z)``

Reply:
In #12062 the problematic line is
```assert allclose(z, funcs.z_at_value(func, fval, bracket=[0.3, 1.0], ztol=1e-12), rtol=2e-11)```
and the warning arises from ``funcs.z_at_value``, not the ``allclose``. So if the test passes, the warning doesn't really matter too much.

Reply:
That being said, I would label this somewhere between a proper fix and temporary workaround. I think we should merge this in and then I'll create an issue to continue digging. Looking at the pytest warning capture options, I can exert more granular control over how the warning is detected and dealt with. The ``recwarn`` is overkill in that it surpasses all warnings. I would rather just filter on the RuntimeWarning. The test does pass, so I'm happy for now.

Reply:
Might be cleaner to just filter the RuntimeWarning if you think it is the only thing causing problem. See https://docs.pytest.org/en/6.2.x/warnings.html#pytest-mark-filterwarnings

But I see that you commented you still cannot reproduce the problem on your side, so I'll keep this open a bit more.